packages/react/src/file-input/FileInput.tsx

```diff
@@ -139,83 +138,76 @@ export const FileInput = forwardRef<HTMLInputElement, FileInputProps>(
 		const fallbackHint =
 			hint || (accept && `Files accepted: ${acceptedFileExtensions}`);
 
+		const required = Boolean(requiredProp);
+		const buttonLabel = `Select ${multiple ? 'files' : 'file'}`;
+		const ariaLabel = [
+			buttonLabel,
+			label,
+			required && 'required',
+			invalid && 'invalid',
+			getSelectedFilesMessage({ fileNames, multiple }),
+		]
+			.filter(Boolean)
+			.join(', ');
```

Comment:
What's the advantage of moving this out of the `Field` and using the raw props rather than the `a11yProps`?

Reply:
Just because we have these props specifically to make things invalid etc. The `a11yprops` are just a byproduct of these